network/dag/signing.go

```diff
@@ -110,5 +123,39 @@ func (d documentSigner) Sign(input UnsignedDocument, signingTime time.Time) (Doc
 	} else {
 		doc.signingKeyID = d.kid
 	}
+	doc.signingAlgorithm = algo
```

Comment:
Together with SignJWS, this seems off. The algo should already be in the JWS header...

Reply:
> I can't fully decipher how the document signing and parsing works, but it looks like there's a layer too many. JWS headers that are copied are a symptom for that.

It was a bit off, in my opinion especially because I returned the algorithm along with the JWS as string. That's fixed now (by just returning a new instance of the signed document, by parsing it). How it's structured;

* lestrrat/jws is used to actually sign JWS's
* Crypto signs generic JWS's using that library and the private keys from storage 
* network/dag/signing.go can network documents using that generic JWS signing using the rules specified by RFC004